### DISCUSSION TITLE
Swagger definitions not being generated when controllers in separate library

### DISCUSSION THREAD

**Discussion by kevmo314** [micronaut-projects/micronaut-core#6518] [Q&A] [answered]
We have a Micronaut application that's built with Bazel. Currently the `main()` and actual application sit under one `java_binary` project and all the `@Controller`s sit in another `java_library` that the application depends on. Micronaut seems to pick up the controllers just fine but it doesn't generate the `swagger.yml` for them. In other words, our `swagger.yml` ends up empty despite the actual routes working correctly.

Can someone point me to where the controllers are registered for `swagger.yml` or where the corresponding annotation processors are in the codebase? I searched a bit but there's quite a lot of abstraction making it a bit challenging to find what I'm looking for. This is a rather unique setup so I realize that I'll likely have to find/solve the issue myself but some pointers to ease my search would be helpful.

**Comment by graemerocher** [comment, marked as answer]
You will probably need to use the include approach https://micronaut-projects.github.io/micronaut-openapi/latest/guide/index.html#openapiinclude

**Reply by kevmo314** [reply]
Thanks! That was it!

**Reply by coltonchrane** [reply]
How would you preserve metadata. For example Kdoc/Javadoc thats used to generate the documentation for swagger ui
